sqlcapture/tests/tests.go

```diff
@@ -239,3 +247,134 @@ func testCatalogPrimaryKeyOverride(ctx context.Context, t *testing.T, tb TestBac
 	})
 	VerifiedCapture(ctx, t, tb, &catalog, &state, "capture2")
 }
+
+func testEncodePgNumeric(t *testing.T) {
+	var testCases = []struct {
```

Comment:
It's good to have a few human-written test cases to validate the encoding looks like we expect, but I think there should be fewer of these, and then a lot more programmatic tests.

I would recommend trimming down this test table, and writing a second test which:
- Generates a list of K random numbers, where K=1,000 or 10,000 or whatever makes the test run in about 1s
- Encodes each of these numbers down to the final `[]byte` FDB serialization
- Decodes the numbers again and checks the result, just to make sure the whole process works end-to-end
- Iterates over every pair of numbers and checks that the ordering of the `[]byte` serialization matches that of the original numbers.

Reply:
Done.